### PR TITLE
[MIG] web_responsive test to 11.0

### LINKED ISSUE #759: [11.0] web_responsive: odoo.define_section is undefined
Current 11.0 builds will all fail due to:

 https://github.com/OCA/web/blob/d62b1b563af449317ab9cada65998df4f0707902/web_responsive/static/tests/js/web_responsive.js#L4

Odoo 11.0 changed the way we write widget tests, and in this change, it removed `odoo.define_section`. That's why we are getting [errors such as this one](https://travis-ci.org/OCA/web/jobs/287908508#L942-L945):

```
` AssertionError: ("TypeError: undefined is not a function (evaluating 'odoo.define_section')\n"
`  ' at http://127.0.0.1:8069/web_responsive/static/tests/js/web_responsive.js:4 '
`  '(in global code)\n'
`  '(leaf frame on top)')
```

We have to fix that, and it would be good to add the instructions to https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0 once discovered how to do it.

The good news is that we have lots of good examples available and some introduction in slides 35-41 of @gde-odoo's [presentation at Odoo Experience 2017](https://www.odoo.com/es_ES/slides/slide/the-odoo-js-framework-492).

To start, it seems you have to just use `odoo.define` to run the tests, but I'm not sure the depth of the changes in that file to adapt to new tests, to be honest...

cc @Tecnativa @lasley @angelmoya

### PR DESCRIPTION
**Pull request by hgminh95**
Fix #759 

@Yajo can you take a look?

### CODE REVIEW

**Review by lasley — CHANGES_REQUESTED** [review]
Neat, QUnit is being exposed directly now. I actually take back some of my crap talking about them changing things then, this is totally for the best.

Can you please increase the tab level on the internals though? This would be a stylistic decision that I'm sure @Yajo and I could go back and forth on endlessly. I made the decision here to have the tab level where it's at though, so please respect that. Otherwise, the diff is quite large and it is hard to understand the changes that are being made.

**Comment by hgminh95** [comment]
Sure, I have changed the style. However, the diff is still quite big as there are some minor changes like `self` become `this`, ...

File: web_responsive/static/tests/js/web_responsive.js [hunk 1]
```diff
@@ -1,278 +1,317 @@
...
+                    .append(
+                        $('<button class="drawer-toggle"><span class="drawer-hamburger-icon">')
+                    )
+                    .append(
+                        $('<nav class="drawer-nav"><ul class="drawer-menu"><li class="drawer-menu-item">')
+                    )
+                    .append(
+                        $('<div class="panel-title" id="appDrawerAppPanelHead">')
+                    )
+            ).append($('<main role="main">'));
```
Comment (yajo) [inline web_responsive/static/tests/js/web_responsive.js]: I think it's better to just bundle all of this appends in the single first `var $el = ...` definition. KISS.

File: web_responsive/static/tests/js/web_responsive.js [hunk 2]
```diff
@@ -1,278 +1,317 @@
...
+
+            this.$clickZone = $('<a class="oe_menu_leaf">');
+
+            this.$secondaryMenu = $('<div><div class="o_sub_menu_content">');
+
+            this.$dropdown = $('<div class="dropdown-scrollable">');
+
+            $el.append(this.$clickZone)
+                .append(this.$secondaryMenu)
+                .append(this.$dropdown);
```
Comment (yajo) [inline web_responsive/static/tests/js/web_responsive.js]: Same about this. It seems easier to read when you drop in the full HTML and then use `$el.find()` to get the specific elements you need IMHO.

File: web_responsive/static/tests/js/web_responsive.js [hunk 3]
```diff
@@ -1,278 +1,317 @@
...
+        linkGrid: function() {
+            for(var i = 0; i < 3; i++){
+                this.drawer.$el.append(
+                    $('<div class="row">').append(
+                        $('<a class="col-md-6" id="a_' + i + '"><span class="app-drawer-icon-app /></a>' +
+                          '<a class="col-md-6" id="b_' + i + '"><span class="app-drawer-icon-app /></a>'
+                          )
+                    )
+                );
+                this.drawer.$appLinks = $('a.col-md-6');
```
Comment (yajo) [inline web_responsive/static/tests/js/web_responsive.js]: Better use a namespaced `$el.find()` instead of whole-page `$()`

**Review by yajo — CHANGES_REQUESTED** [review]
Beautiful

**Comment by hgminh95** [comment]
Thanks for your review. I have updated the code.

**Review by lasley — APPROVED** [review]
Thank you @hgminh95 - absolutely amazing

File: web_responsive/static/tests/js/web_responsive.js [hunk 4]
```diff
@@ -1,278 +1,318 @@
...
-                .append(
-                    $('<nav class="drawer-nav"><ul class="drawer-menu"><li class="drawer-menu-item">')
-                )
+
+    var core = require('web.core');
+    var responsive = require('web_responsive');
+
+    QUnit.module('web_responsive', {
+        beforeEach: function() {
+            var $el = $('<div class="drawer drawer--left">')
```
Comment (yajo) [inline web_responsive/static/tests/js/web_responsive.js]: Still all of these appends seem overly complex. Can't you write normal HTML and put it into this first call?

**Review by yajo — CHANGES_REQUESTED** [review]

File: web_responsive/static/tests/js/web_responsive.js (hunk 4 above)
Comment (lasley) [reply web_responsive/static/tests/js/web_responsive.js]: I'm not saying it won't work, but I really hope I wouldn't have done something this incredibly complex and stupid for no reason. (this is just following the pattern of my original code)

**Comment by hgminh95** [comment]
Done :smile:

File: web_responsive/static/tests/js/web_responsive.js (hunk 4 above)
Comment (lasley) [reply web_responsive/static/tests/js/web_responsive.js]: So yeah guess I did do it for no reason. Thanks @hgminh95!

**Review by yajo — APPROVED** [review]

**Pull request merged**